### PR TITLE
Added TTF_FontStyleFlags and TTF_HintingFlags

### PR DESCRIPTION
**Pull request by edwardgushchin**
Added TTF_FontStyleFlags and TTF_HintingFlags.

### CODE REVIEW

File: src/SDL_ttf.c [hunk 1]
```diff
@@ -247,7 +247,7 @@ struct TTF_Font {
     int lineskip;
 
     // The font style
-    int style;
+    Uint32 style;
```
Comment (slouken) [inline src/SDL_ttf.c:250]: Shouldn't this be TTF_FontStyleFlags? And the same for prev_style below?

File: include/SDL3_ttf/SDL_ttf.h [hunk 2]
```diff
@@ -447,13 +458,23 @@ extern SDL_DECLSPEC bool SDLCALL TTF_SetFontOutline(TTF_Font *font, int outline)
...
+ *
+ * \sa TTF_SetFontHinting
+ * \sa TTF_GetFontHinting
  */
-#define TTF_HINTING_NORMAL          0
-#define TTF_HINTING_LIGHT           1
-#define TTF_HINTING_MONO            2
-#define TTF_HINTING_NONE            3
-#define TTF_HINTING_LIGHT_SUBPIXEL  4
+typedef Uint32 TTF_HintingFlags;
```
Comment (slouken) [inline include/SDL3_ttf/SDL_ttf.h:471]: Since these values are exclusive, maybe this should be an enum?

Comment (edwardgushchin) [reply include/SDL3_ttf/SDL_ttf.h:471]: Yeah, you're right.

File: src/SDL_ttf.c (hunk 1 above)
Comment (edwardgushchin) [reply src/SDL_ttf.c:250]: I've been thinking about it. Wasn't sure it wouldn't break anything.

File: include/SDL3_ttf/SDL_ttf.h [hunk 3]
```diff
@@ -447,13 +458,24 @@ extern SDL_DECLSPEC bool SDLCALL TTF_SetFontOutline(TTF_Font *font, int outline)
 extern SDL_DECLSPEC int SDLCALL TTF_GetFontOutline(const TTF_Font *font);
 
 /**
- * Hinting flags
+ * Hinting flags for TTF (TrueType Fonts)
+ *
+ * These flags specify the level of hinting to be applied to the font rendering.
```
Comment (slouken) [inline include/SDL3_ttf/SDL_ttf.h:463]:
```suggestion
 * This enum specifies the level of hinting to be applied to the font rendering.
```

**Comment by slouken** [comment]
Merged, thanks!

**Pull request merged**